torch/csrc/autograd/record_function_ops.cpp

```diff
@@ -14,6 +14,47 @@ namespace torch {
 namespace autograd {
 namespace profiler {
 
+// Holder of RecordFunction, used to store the state of a RecordFunction
+// object to record the enter and exit event for profiler.
+struct RecordFunctionHolder : torch::CustomClassHolder {
+  std::unique_ptr<at::RecordFunction> record_function_;
+
+  RecordFunctionHolder() {
+    record_function_ = std::make_unique<at::RecordFunction>(at::RecordScope::USER_SCOPE);
+  }
+  void enter(const std::string& name) {
+    if (record_function_ == nullptr) {
+      LOG(ERROR) << "record_function_ should never be NULL";
```

Comment:
Let's use `TORCH_CHECK` so that we throw an informative error here.

Reply:
Done